native/cocos/bindings/jswrapper/PrivateObject.h

```diff
@@ -94,7 +94,7 @@ class SharedPrivateObject final : public TypedPrivateObject<T> {
 
     constexpr bool isSharedPtr() const override { return true; }
 
-    void *getRaw() const override { return _data.get(); }
+    void *getRaw() const override { return (void *)_data.get(); }
```

Comment:
`::getRaw` in `CCSharedPtrPrivateObject` and `RawRefPrivateObject` should also do such cast